### PR TITLE
A Bunch of small things to support this project! 

### PR DESCRIPTION
**Pull request by Jomyy**
new instance in sheet and ui cleanups like add button as icon button and real searchbar for the instances, also download libs, assets and Minecraft jar on creation of new instance

### CODE REVIEW

**Comment by Jomyy** [comment]
Bildschirm­foto 2022-12-15 um 19 54 39
Bildschirm­foto 2022-12-15 um 19 55 39

File: InnateMC/ContentView.swift [hunk 1]
```diff
@@ -29,49 +29,55 @@ struct ContentView: View {
...
-                ForEach(viewModel.instances) { instance in
-                    if ((!starredOnly || instance.isStarred) && (searchTerm.isEmpty || instance.checkMatch(searchTerm))) {
-                        NavigationLink(destination: {
-                            InstanceView(instance: instance)
-                                .padding(.top, 10)
-                        }, label: {
-                            InstanceNavigationLink(instance: instance)
-                        })
+       
+        if #available(macOS 12.0, *) {
```
Comment (BoogieMonster1O1) [inline InnateMC/ContentView.swift:33]: why does this need mac os 12? I run mac os 11 and `NavigationView` works with mac os 11

File: InnateMC/Instance/InstanceLaunchView.swift [hunk 2]
```diff
@@ -26,6 +26,12 @@ struct InstanceLaunchView: View {
         VStack {
             HStack {
                 Button(action: {
+                    print(instance.getPath().absoluteString)
+                    do{
+                        
+                    }catch{
```
Comment (BoogieMonster1O1) [inline InnateMC/Instance/InstanceLaunchView.swift:32]:
```suggestion
                    } catch {
```

File: InnateMC/Instance/InstanceLaunchView.swift [hunk 3]
```diff
@@ -26,6 +26,12 @@ struct InstanceLaunchView: View {
         VStack {
             HStack {
                 Button(action: {
+                    print(instance.getPath().absoluteString)
+                    do{
```
Comment (BoogieMonster1O1) [inline InnateMC/Instance/InstanceLaunchView.swift:30]:
```suggestion
                    do {
```

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]

File: InnateMC/ContentView.swift (hunk 1 above)
Comment (Jomyy) [reply InnateMC/ContentView.swift:33]: Oops left it in from when I was trying to use the . searchable thingy

File: InnateMC/Instance/InstanceRuntimeView.swift [hunk 4]
```diff
@@ -22,7 +22,10 @@ import SwiftUI
 struct InstanceRuntimeView: View {
     let columns = [GridItem(.fixed(200), alignment: .trailing), GridItem(.flexible(), alignment: .leading)]
     var instance: Instance
-    
+    @AppStorage("innatemc.javaExecutable") private var javaExecutable: String = "java"
+    @AppStorage("innatemc.minMemory") private var minMemory: Int = 1024
+    @AppStorage("innatemc.maxMemory") private var maxMemory: Int = 1024
```
Comment (BoogieMonster1O1) [inline InnateMC/Instance/InstanceRuntimeView.swift:27]: This is instance-specific, and shouldn't be stored with AppStorage

**Pull request merged**